lib/concurrent/parallel.rb

```diff
@@ -0,0 +1,91 @@
+require 'concurrent/atomics'
+require 'concurrent/configuration'
+require 'concurrent/executors'
+
+module Concurrent
+
+  class Parallel < SimpleDelegator
+    module EnumerableInterface
+      module_function
+
+      def use(runner, methods)
+        enumerable_methods = Enumerable.instance_methods.map(&:to_s)
+        enumerable_methods << 'each'
+        methods.each do |method|
+          next unless enumerable_methods.include?(method)
+          class_eval <<-RUBY
+          def #{method}(*args, &block)
+            #{runner}(:#{method}, *args, &block)
+          end
+          RUBY
+        end
+      end
+
+      use :run_in_threads, %w(
+        all? any? count detect find find_index max_by min_by minmax_by none?
+        one? partition
+      )
+
+      use :run_in_threads_return_original, %w(
+        cycle each each_cons each_entry each_slice each_with_index enum_cons
+        enum_slice enum_with_index reverse_each zip
+      )
+
+      use :run_in_threads_return_parallel, %w(
```

Comment:
I agree with you @jdantonio.  I'm a big fan of explicit behavior so I think the second example would be a better.